### DISCUSSION TITLE
Dynamically import JSON will fail in production mode

### DISCUSSION THREAD

**Discussion by ChiuMungZitAlexander** [vercel/next.js#63233] [Help]
### Summary

Inside a route handler `/api/route.ts`, I want to import different JSON files by the environment variables. I did as below
```
const { default: map }: { default: Record<string, Data> } = await import(`path-to-json/${env}.json`);
```
This is working perfectly in dev mode (`next dev`). Also, the project can be compiled successfully(`next build`).

However, when running production(`next start`), this JSON will be an **empty object** `{}`. It will NOT throw any error of path resolving or else.

I also tried using assertion as below, but got the same result.
```
const { default: map }: { default: Record<string, Data> } = await import(`path-to-json/${env}.json`, { with: { type: 'json' } });
```

Even though I solved it as below, I wonder why `await import` will fail.
```
const json = await fs.readFile(
  process.cwd() + `path-to-json/${env}.json`,
  'utf-8',
);
const data: Record<string, Data> = JSON.parse(json);
```

**Comment by FranciscoCaetano88** [comment]
I had to install the json-loader plugin and add a rule to the next.config to deal with json files differently for some specific paths and now it works as intended. Example:
```js
config.module.rules.push({
        test: /\.json$/,
        type: 'javascript/auto',
        use: 'json-loader',
        include: pathToSpecificFiles
});
```
